packages/core-data/src/private-selectors.ts

```diff
@@ -93,3 +98,243 @@ export function getEntityRecordPermissions(
 ) {
 	return getEntityRecordsPermissions( state, kind, name, id )[ 0 ];
 }
+
+export const getRelatedEditedEntityRecordsByTemplate = createRegistrySelector(
```

Comment:
I'm pretty sure the logic can be improved and simplified 🙂 I just got something working to showcase the need we have. Any guidance is more than welcome, though.

Reply:
It' very unclear to me what this function does and how it relates to the current PR?

Reply:
We still need to polish the PR and the description 😄 I'll try to sum up the use case.

* We want to be able to connect custom fields to block from a template, where there is no post ID because it applies to all blocks.
* This means we need to access all the registered fields for a postType.
* We thought it could be useful to expose those in the `types` REST API endpoint.
* However, depending on the template, the custom fields might be different. If I am in the movie CPT template, I want to access the custom fields registered for the "movie" postType. If I am in an actor CPT template, the ones for actor.
* With this function, I was trying to infer the entity each template is connected to based on its slug. Similar to how [template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/) works.

I didn't think too much if this is the best approach, I was just trying to use it to illustrate the use case.

Reply:
Given that you can assign the same template to multiple post types, I don't see a way to reliably guess the post type without a postId. I feel like we should avoid anything smart like that. Are there other alternatives?

Reply:
> Given that you can assign the same template to multiple post types

My idea for the templates that can work for multiple post types was to show the custom fields that are available for any post type.

> Are there other alternatives?

I still have to think more about it. I thought using the slug could work this way, at least as a first step:
* If a template is specific to a post type, use the custom fields for that post type. Ex: [`object_subtype`](https://developer.wordpress.org/reference/functions/register_meta/#parameters) equals `movie`.
* If a template is generic to any post type, use the custom fields registered for any post type. [`object_subtype`](https://developer.wordpress.org/reference/functions/register_meta/#parameters) not defined.

Reply:
> I still have to think more about it. I thought using the slug could work this way, at least as a first step:
> If a template is specific to a post type, use the custom fields for that post type. Ex: [object_subtype](https://developer.wordpress.org/reference/functions/register_meta/#parameters) equals movie.
> If a template is generic to any post type, use the custom fields registered for any post type. [object_subtype](https://developer.wordpress.org/reference/functions/register_meta/#parameters) not defined.

Why aren't we using the "postType" from the block context rather than trying to guess the post type from the template?

Imagine, you're editing "archive of some CPT A", why would you want to show the custom fields for CPT A in that case? it's an archive page which means it renders multiple elements of that CPT which means we don't really know which post ID to bind to, so showing the right fields is not useful no? Maybe I'm missing something but it seems that this complexity is not really useful in terms of actual usage.

Reply:
> Why aren't we using the "postType" from the block context rather than trying to guess the post type from the template?

If we are in a template, the `postType` from the block context is `undefined` according to my tests. There is no `postId` or `postType` because it is actually a template.

> Imagine, you're editing "archive of some CPT A", why would you want to show the custom fields for CPT A in that case? it's an archive page which means it renders multiple elements of that CPT which means we don't really know which post ID to bind to, so showing the right fields is not useful no?

If you are using are editing a template for an archive, we wouldn't show the custom fields of a CPT. We would show the taxonomy custom fields if that's supported at some point.

> Maybe I'm missing something but it seems that this complexity is not really useful in terms of actual usage.

I am not convinced about the solution either, although it seems is how classic themes templates work with file names.

I believe the use case of accessing the custom fields from a template is valid though. I'll try to think about different approaches.

Reply:
> If you are using are editing a template for an archive, we wouldn't show the custom fields of a CPT. We would show the taxonomy custom fields if that's supported at some point.

I don't understand, why would you show taxonomy custom fields in the "archive of some CPT A"?

Reply:
> why would you show taxonomy custom fields in the "archive of some CPT A"?

Sorry, I thought you were talking about a taxonomy template.

I will remove the existing selector because it is not clear if it should exist at all. I believe we can start simpler by moving the logic to the `post-meta` source definition, still reading the slug, at least until we figure out a better option. Right now, the UI is under an experimental flag, so I don't think it should be a problem.

For context, there are other cases where code is relying on the slug: [link 1](https://github.com/WordPress/gutenberg/blob/a43e39194f25d39e69426b15a2b9036022f301d3/packages/block-library/src/post-template/edit.js#L114-L121) or [link 2](https://github.com/WordPress/gutenberg/blob/a43e39194f25d39e69426b15a2b9036022f301d3/packages/block-library/src/query-title/use-archive-label.js#L30-L33).


I'll try to make a commit to explain what I mean.

Reply:
I moved the logic to the "Post meta" source and removed the previous selector. I think it is safer for now to keep the logic there, but I still couldn't come up with a way to avoid inferring the post type from the slug as done [here](https://github.com/WordPress/gutenberg/pull/64072/files#diff-cf4447b4134dbf24fa2819703999f0131449fb79a0611bd544c5e05fd80aa36bR107-R109). I'm still not convinced about the code, so I'll keep thinking about it.